frontend/survey-project/src/app/shared/services/user.service.ts

```diff
@@ -45,4 +53,48 @@ export class UserService {
                    );
     }
 
+    logoutUser(): Observable<LogoutResponse> {
+        const url = `${this.baseUrl}/logout`; 
+        return this.httpClient.post<LogoutResponse>(url, {})
+                   .pipe(
+                        catchError((e: HttpErrorResponse) => {
+                            return throwError(() => e.error);
```

Comment:
It's not necessary to use arrow function to pass a simple argument, you could just written **return throwError(e.error);**